### ISSUE TITLE
[BUG] await AnalyzeBatchDocumentsAsync(WaitUntil.Completed, ..) hangs indefinitely even when operation is completed

### ISSUE THREAD

**Issue by wissembs** [Azure/azure-sdk-for-net#46931]
### Library name and version

Azure.AI.DocumentIntelligence 1.0.0-beta.3

### Describe the bug

I'm experiencing issues getting operation Status after using AnalyzeBatchDocumentsAsync() both with WaitUntil.Completed or WaitUntil.Started.
Specifically, my attempts to retrieve the AnalyzeBatchResult are resulting in indefinite waits, despite the operation being completed and the result files being successfully stored in the result container.

1. Using WaitUntil.Completed:

- The await call waits indefinitely and never returns AnalyzeBatchResult, even though the result files are present in the ResultContainer. When checking the operation status separately using Postman, it shows AnalyzeBatchResultOperation with a status of "completed."

2. Using WaitUntil.Started followed by WaitForCompletionAsync():

- The same indefinite wait issue occurs, even though the result files are correctly generated and saved. Again, the status checked through Postman indicates that the operation is completed.

3. Using WaitUntil.Started and attempting to retrieve documentOperation.Id:

- Attempting to access documentOperation.Id throws a System.NotSupportedException with the message: Specified method is not supported.

### Expected behavior

- AnalyzeBatchDocumentsAsync should return the AnalyzeBatchResult once the operation is completed.

- WaitForCompletionAsync() should correctly wait and return the result.

- documentOperation.Id shoud be accessible, to try checking the status through http call to [Get Analyze Batch Result](https://learn.microsoft.com/en-us/rest/api/aiservices/document-models/get-analyze-batch-result?view=rest-aiservices-v4.0%20(2024-07-31-preview)&tabs=HTTP#analyzebatchresultoperation)

- Having a method in DocumentIntelligenceClient for [Get Analyze Batch Result](https://learn.microsoft.com/en-us/rest/api/aiservices/document-models/get-analyze-batch-result?view=rest-aiservices-v4.0%20(2024-07-31-preview)&tabs=HTTP#analyzebatchresultoperation)

### Actual behavior

The methods wait indefinitely despite the operation being completed and results being available in the Azure Blob Storage container.
I have confirmed that the operation completes by checking [Get Analyze Batch Result](https://learn.microsoft.com/en-us/rest/api/aiservices/document-models/get-analyze-batch-result?view=rest-aiservices-v4.0%20(2024-07-31-preview)&tabs=HTTP#analyzebatchresultoperation) with Postman, and it returns AnalyzeBatchResultOperation with the status "completed."

### Reproduction Steps

```csharp
public async Task GetDocumentTextAsync(string blobPrefix)
{
    Uri sourceContainerUri = new($"https://{_fileStorageOptions.AccountName}.blob.core.windows.net/{_fileStorageOptions.SourceContainerName}/");
    Uri resultContainerUri = new($"https://{_fileStorageOptions.AccountName}.blob.core.windows.net/{_fileStorageOptions.ProcessedContainerName}/");

    AnalyzeBatchDocumentsContent analyzeBatchRequest = new AnalyzeBatchDocumentsContent(resultContainerUri)
    {
        AzureBlobSource = new AzureBlobContentSource(sourceContainerUri)
        {
            Prefix = blobPrefix,
        },
        ResultPrefix = blobPrefix,
        OverwriteExisting = true,
    };

    try
    {
        // Using WaitUntil.Completed should return AnalyzeBatchResult, but the operation hangs indefinitely
        // Operation<AnalyzeBatchResult> documentOperation = await _documentClient.AnalyzeBatchDocumentsAsync(WaitUntil.Completed, "prebuilt-layout", analyzeBatchRequest: analyzeBatchRequest);
        // AnalyzeBatchResult responseData = documentOperation.Value;
        // _logger.LogInformation("Operation finished: SucceededCount = {SucceededCount}", responseData.SucceededCount);

        // Trying WaitUntil.Started, then WaitForCompletionAsync() also hangs indefinitely
        Operation<AnalyzeBatchResult> documentOperation = await _documentClient.AnalyzeBatchDocumentsAsync(WaitUntil.Started, "prebuilt-layout", analyzeBatchRequest: analyzeBatchRequest);
        
        // documentOperation.Id throws a System.NotSupportedException with the message: Specified method is not supported.
        // _logger.LogInformation("Operation Started: {OperationId}", documentOperation.Id);

        Response<AnalyzeBatchResult> response = await documentOperation.WaitForCompletionAsync();
        AnalyzeBatchResult value = response.Value;
        _logger.LogInformation("Operation finished: SucceededCount = {SucceededCount}", value.SucceededCount);
    }
    catch (RequestFailedException ex)
    {
        _logger.LogError(ex, ex.Message);
    }
}
```

### Environment
SDK .NET :
 Version:           8.0.400
 Commit:            36fe6dda56
 Workload version:  8.0.400-manifests.b78e89c6
 MSBuild version:   17.11.3+0c8610977

Environnement d'exécution :
 OS Name:     Windows
 OS Version:  10.0.22631
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.400\

Azure.AI.DocumentIntelligence 1.0.0-beta.3
Aspire.Hosting.AppHost 8.2.1

**Comment by jsquire** [comment]
/Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by wissembs** [comment]
Hi @kinelski @jsquire ! Just checking in to see if there are any updates on this issue or if additional information would be helpful for understanding the problem. Any insights or estimated timelines would be greatly appreciated !

**Comment by kinelski** [comment]
@wissembs, thank you for reaching out and apologies for the delay in getting back to you.

Both issues you were experiencing (the indefinite delay and the `NotSupportedException` when accessing the ID of the operation) were confirmed to be bugs in the .NET SDK and have been fixed in our latest release: 1.0.0. You should stop seeing them when you update.

Please be aware that some breaking changes have been introduced in the batch API, so feel free to check our CHANGELOG for an exhaustive list of changes: [CHANGELOG](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/documentintelligence/Azure.AI.DocumentIntelligence/CHANGELOG.md#breaking-changes-1)
